src/main/scala/foo/MainServer.scala

```diff
@@ -80,7 +80,7 @@ class HttpServer[F[_]: ConcurrentEffect : ContextShift : Timer : Par](implicit v
     }
 
     private def block[A](thunk: => A, timeout: FiniteDuration): F[A] =
-      contextShift.evalOn(blockingEc)(cats.effect.IO(thunk).timeout(timeout)).to[F]
+      contextShift.evalOn(blockingEc)(cats.effect.IO(thunk)).timeout(timeout).to[F]
```

Comment:
This one is subtle: the `timeout` shifts it back.  As written, it's sleeping on the `contextShift`'s execution context instead of the `blockingEc`.